libraries/AP_HAL_Linux/I2CDevice.cpp

```diff
@@ -217,7 +217,7 @@ bool I2CDevice::read_registers_multiple(uint8_t first_reg, uint8_t *recv,
         };
 
         int r = -EINVAL;
-        unsigned retries = _retries;
+        uint8_t retries = _retries;
```

Comment:
unsigned is a CPU word... so often the underlying storage is uint8_t/uint16_t but then we use an unsigned to do math since it's faster to do it that way.  You can check this by comparing the generated assembly, but just looking at the binary size is a way to see you are actually doing more things:

Before:
```
Target               Text     Data  BSS    Total 
bin/arducopter-quad  1186814  2360  37680  1226854
```

After:
```
Target               Text     Data  BSS    Total 
bin/arducopter-quad  1186862  2360  37680  1226902
```